packages/opentelemetry-tracing/src/Tracer.ts

```diff
@@ -87,7 +87,9 @@ export class Tracer implements api.Tracer {
     const attributes = sanitizeAttributes(options.attributes);
     // make sampling decision
     const samplingResult = this._sampler.shouldSample(
-      context,
+      options.root
+        ? api.setSpanContext(context, api.INVALID_SPAN_CONTEXT)
```

Comment:
I think `context.deleteValue(SPAN_KEY)` would be the correct approach here. I have seen several places around where checks like `if (getSpan(context) != null)` are done to detect presence of a parent.

Reply:
Spec states that "The Span's SpanContext may be invalid to indicate a root span." and there is no function to remove span from context currently. We don't have the SPAN_KEY outside of the API

Reply:
ok.

Maybe we should add `deleteSpan(context)` (and `deleteBaggage(context)`) at some time?

Reply:
> Maybe we should add deleteSpan(context) (and deleteBaggage(context)) at some time?

totally make sense